src/mixer/basetrackplayer.cpp

```diff
@@ -335,3 +345,30 @@ void BaseTrackPlayerImpl::setupEqControls() {
     m_pSpeed = new ControlObjectSlave(group, "rate", this);
     m_pPitchAdjust = new ControlObjectSlave(group, "pitch_adjust", this);
 }
+
+void BaseTrackPlayerImpl::slotPassthroughEnabled(double v) {
+    bool configured = m_pInputEnabled->toBool();
+    bool passthrough = v > 0.0;
+
+    // Warn the user if they try to enable passthrough on a player with no
+    // configured input.
+    if (!configured && passthrough) {
+        m_pPassthroughEnabled->set(0.0);
```

Comment:
Hm, actually the BaseTrackPlayer derivative does not own the CO so a valueChangeRequest isn't quite right here and I don't think the EngineXXX object should be emitting signals, as we should be reducing the use of signals/slots in the engine.

Since this is a refactoring I'll leave this as a TODO. The existing code in HEAD already behaves this way. A proper fix will be possible when the BaseTrackPlayer derivative owns the CO API for the engine object.